epictrack-web/src/components/workPlan/issues/IssueAccordion/Details.tsx

```diff
@@ -116,20 +120,40 @@ const IssueDetails = ({ issue }: { issue: WorkIssue }) => {
               </If>
 
               <Grid item>
-                <Button
-                  variant="text"
-                  startIcon={<PencilEditIcon />}
-                  sx={{
-                    backgroundColor: "inherit",
-                    borderColor: "transparent",
-                  }}
-                  onClick={() => {
-                    setUpdateToEdit(latestUpdate);
-                    setShowIssuesForm(true);
-                  }}
-                >
-                  Edit
-                </Button>
+                <If condition={latestUpdate.is_approved}>
+                  <Restricted allowed={[ROLES.EXTENDED_EDIT]}>
```

Comment:
This would hide the wrapped component. You could always pass it props in the  errorProps object. errorProps: {{ disabled: true}}. This would pass the disabled prop to the child